src/dom-to-image-more.js

```diff
@@ -1010,45 +1012,147 @@
     let removeDefaultStylesTimeoutId = null;
     let tagNameDefaultStyles = {};
 
-    function getDefaultStyle(tagName) {
-        if (tagNameDefaultStyles[tagName]) {
-            return tagNameDefaultStyles[tagName];
+    const ascentStoppers = [
+        // these come from https://developer.mozilla.org/en-US/docs/Web/HTML/Block-level_elements
+        'ADDRESS',
+        'ARTICLE',
+        'ASIDE',
+        'BLOCKQUOTE',
+        'DETAILS',
+        'DIALOG',
+        'DD',
+        'DIV',
+        'DL',
+        'DT',
+        'FIELDSET',
+        'FIGCAPTION',
+        'FIGURE',
+        'FOOTER',
+        'FORM',
+        'H1',
+        'H2',
+        'H3',
+        'H4',
+        'H5',
+        'H6',
+        'HEADER',
+        'HGROUP',
+        'HR',
+        'LI',
+        'MAIN',
+        'NAV',
+        'OL',
+        'P',
+        'PRE',
+        'SECTION',
+        'SVG',
+        'TABLE',
+        'UL',
+        // these are ultimate stoppers in case something drastic changes in how the DOM works
+        'BODY',
+        'HEAD',
+        'HTML',
+    ];
+
+    function getDefaultStyle(sourceElement) {
+        const tagHierarchy = computeTagHierarchy(sourceElement);
+        const tagKey = tagHierarchy.join('>'); // it's like CSS
```

Comment:
This adds a cost for generalised tags like span and i which I suspect is exponential - $O(c^N)$ where $c$ is average hierarchy depth for that tag. As we're constructing against tag hierarchy in the DOM, where most developers religiously follow "permitted (non-normative) context", we should be able to get away with caching by tag name, or reducing the key to a maximum of 2 - `tagName` or `ascentStopper tagName`. Need to protect against #70 regression here

Reply:
The ideal way to stress this algo in performance testing is to go to a complex Wikipedia article like https://en.wikipedia.org/wiki/JavaScript?useskin=vector-2022

If you have qualms about overly simplifying the caching logic, we can loop in the original author and see what they think RE: perf cost-benefit

Reply:
Yep, that would do it _IFF_ the assertion is that `TD>TR>TBODY>TABLE` _results_ in the same default styles as `TD>TR>TFOOT>TABLE` and other similar, but not sufficiently dissimilar fragments (both would end up with the `tagKey` of `'TD>TABLE'`. I'm going to see what it looks like. Do you have a performance test-jig for doing arbitrary pages? Might be a nice jsfiddle

Reply:
I've updated for the option to be `strict` or `relaxed` in case someone breaks. We default to `strict`